docs/node/network-migration.mdx

```diff
@@ -0,0 +1,92 @@
+---
+sidebar_position: 6
+sidebar_label: "Network Migrations"
+id: network-migrations
+title: Network Migrations (Hard forks)
+description: Network Migrations
+slug: /node/network-migrations
+---
+This document provides a detailed guide on how to establish a new network that integrates existing data from a previous network, using nodes that run on the updated version of the `kwild` binaries. This approach is necessary when seamless upgrades to new versions are not feasible due to significant, breaking changes that affect consensus mechanisms and transaction replays.
+
+## Network Migrations
+
+Follow the steps below to upgrade the network to a new version of the `kwil` binaries:
+
+### Create Snapshot
+
+Stop the network by shutting down all the nodes in the network. Then, use the `kwil-admin snapshot` tool to take a snapshot of the final state of the Kwild database. For more details, refer to the [create database snapshots](/docs//ref/kwil-admin/snapshot/create) documentation. This tool connects directly to the database to capture its state and does not require the `kwild` process to be running.
```

Comment:
Do they have to stop nodes? I don't think they have to stop all of them, but do they need to stop the one they are snapshotting?

Reply:
Yeah, we are doing a network migration here. I asked Jon to have a feature in Consensus Upgrades which will stop the network from mining blocks, that way, we can do coordinated stops and can get the snapshot of the latest state. and start a new network without losing any more transactions

Reply:
I see. I think there might be a better way for us to handle this that can allow for them to have less downtime. Will share in a bit, but the tldr; is using oracles to listen to the old network.

Reply:
I'm curious about the proposed solution.  If I were coordinating the death and rebirth of a network, I would expect to have all validators in on it.  Therefore, the last validator to hit CTRL+C gets to make the genesis data file.  Regardless, a pre-programmed halting height is a minor improvement that I'd support.

However, I still don't like we have to support this kind of migration.  So, I hope that whatever oracle solution we devise isn't complex, otherwise we might as well just not brick networks with incompatible upgrades in the first place.  The need to keep the old network running and accessible by clients is undesirable complexity.

Is there precedent for launching a new network that is linked to the old one, but with new rules, where the intent is to migrate all use to the new one?